librz/analysis/arch/tricore/tricore_il.inc

```diff
@@ -1,23 +1,137 @@
 // SPDX-FileCopyrightText: 2023 curly <billow.fun@gmail.com>
 // SPDX-License-Identifier: LGPL-3.0-only
-#include <rz_il.h>
+
 #include <rz_il/rz_il_opbuilder_begin.h>
+
+#define LOAD32(addr) LOADW(32, addr)
+
+static inline RzILOpPure *bits(RzILOpBitVector *bv, ut32 i, ut32 n) {
+	return LOGAND(SHIFTR0(bv, U32(i)), U32((1 << n) - 1));
+}
+static inline RzILOpPure *bit(RzILOpBitVector *bv, ut32 i) {
+	return bits(bv, i, 1);
+}
+static inline RzILOpEffect *setg_bits(const char *name, ut32 i, ut32 n, RzILOpPure *x) {
+	return SETG(name, LOGOR(LOGAND(VARG(name), U32(~(((1 << n) - 1) << i))), SHIFTL0(x, U32(i))));
+}
+static inline RzILOpEffect *setg_bit(const char *name, ut32 i, RzILOpBitVector *x) {
+	return setg_bits(name, i, 1, x);
+}
+
+#define REG_FIELD(regname, fieldname, i, n) \
+	static inline RzILOpPure *regname##_##fieldname() { \
+		return bits(VARG((#regname)), (i), (n)); \
+	} \
+	static inline RzILOpEffect *set_##regname##_##fieldname(RzILOpPure *x) { \
+		return setg_bits((#regname), (i), (n), (x)); \
+	}
+
+#define VAR_PSW VARG("PSW") ///<  Program Status Word Register
+REG_FIELD(PSW, C, 31, 1)
+REG_FIELD(PSW, V, 30, 1)
+REG_FIELD(PSW, SV, 29, 1)
+REG_FIELD(PSW, AV, 28, 1)
+REG_FIELD(PSW, SAV, 27, 1)
+REG_FIELD(PSW, PRS, 12, 2)
+REG_FIELD(PSW, IO, 10, 2)
+REG_FIELD(PSW, IS, 9, 1)
+REG_FIELD(PSW, GW, 8, 1)
+REG_FIELD(PSW, CDE, 7, 1)
+REG_FIELD(PSW, CDC, 0, 7)
+
+#define VAR_FCX VARG("FCX") ///< Previous Context Pointer Register.
+REG_FIELD(FCX, FCXS, 16, 4)
+REG_FIELD(FCX, FCXO, 0, 15)
+
+#define VAR_PCXI VARG("PCXI") ///< Previous Context Information Register.
+REG_FIELD(PCXI, PCPN, 24, 8)
```

Comment:
Note, that PCXI structure depends on the Tricore version used: https://gitlab.com/qemu-project/qemu/-/issues/1453

Reply:
> Note, that PCXI structure depends on the Tricore version used: [gitlab.com/qemu-project/qemu/-/issues/1453](https://gitlab.com/qemu-project/qemu/-/issues/1453)

Do we need to support all versions, or just a limited few?

Reply:
@imbillow all. Also, it might be necessary for the future ISA versions as well, since they might change something in the future again. Thus, we should build the ISA versioning in.  See, for example, how it's handled in `librz/analysis/p/analysis_mips_cs.c` (search for `analysis->cpu`).

Reply:
For testing purposes, unlike in the code, it's probably better to limit ourselves to the two latest major ISA versions though - 1.3.1 and 1.6.2